src/components/expression/expression-renderer.js

```diff
@@ -630,7 +629,7 @@ genotet.ExpressionRenderer.prototype.dataLoaded = function() {
 
 /** @inheritDoc */
 genotet.ExpressionRenderer.prototype.dataReady = function() {
-  return this.data.tfaData != null;
+  return this.data.tfa.tfaData != null;
```

Comment:
avoid the same prefix in multi-level names. "this.data.tfa" is preferred to be a flatten object.
```js
this.data.tfa = {
  // whatever the server returns
  // whatever you need to store additionally (e.g. fileName)
}
```